src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -559,11 +585,22 @@ public void CancelAsynchronously()
 
             private void CancelSynchronously()
             {
-                if (TrySetCanceled())
+                // AsyncTaskMethodBuilder doesn't give us Try* methods, and the Set methods may throw if the task
+                // is already completed. The belief is that the race is somewhere between rare to impossible, and
+                // so we'll do a quick check to see if the task is already completed or otherwise just give it a shot
+                // and catch it if it fails
+                if (_taskBuilder.Task.IsCompleted)
+                {
+                    return;
+                }
+
+                try
+                {
+                    _taskBuilder.SetException(new OperationCanceledException(_cancellationToken));
```

Comment:
Got it, that part was elided from the diff.  This code just looked like it fit naturally in the body of the method it was called from in the diff.